CMakeLists.txt

```diff
@@ -28,6 +28,25 @@ if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
     if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.1)
         message(FATAL_ERROR "g++ version must be at least 5.1!")
     endif()
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+    include_directories(BEFORE "/opt/intel/include")
```

Comment:
Please do not hard code any path's. It could be that the compiler is installed somewhere else even if this path is the default path.